pkg/sql/catalog/table_elements.go

```diff
@@ -152,9 +176,9 @@ type Index interface {
 	GetSharded() catpb.ShardedDescriptor
 	GetShardColumnName() string
 
-	IsValidOriginIndex(originColIDs descpb.ColumnIDs) bool
-	IsHelpfulOriginIndex(originColIDs descpb.ColumnIDs) bool
-	IsValidReferencedUniqueConstraint(referencedColIDs descpb.ColumnIDs) bool
+	// IsValidOriginIndex returns whether the index can serve as an origin index
+	// for a foreign key constraint.
+	IsValidOriginIndex(fk ForeignKeyConstraint) bool
```

Comment:
side question: I don't understand what purpose this `IsValidOriginIndex` serves. Why would we care whether the originColumns of an FK is a prefix of some index? I only know that we need to care about the referencedColumns of an FK bc we require uniqueness on those column(s).